src/main/java/nl/mpdev/backend_spring_boot_tech_it_easy_controller/controllers/TelevisionsController.java

```diff
@@ -3,76 +3,67 @@
 import nl.mpdev.backend_spring_boot_tech_it_easy_controller.exceptions.RecordNotFoundException;
 import nl.mpdev.backend_spring_boot_tech_it_easy_controller.exceptions.StringTooLongException;
 import nl.mpdev.backend_spring_boot_tech_it_easy_controller.models.Television;
+import nl.mpdev.backend_spring_boot_tech_it_easy_controller.services.TelevisionService;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 @RestController
 @RequestMapping("/api/v1")
 public class TelevisionsController {
 
-  List<Television> televisionDataBase = new ArrayList<>();
+//  List<Television> televisionDataBase = new ArrayList<>();
 
-  @GetMapping("/televisions")
-  public ResponseEntity<List<Television>> getAllTelevisions() {
-//    Map<String, String> response = new HashMap<>();
-//    response.put("television 1","Toshiba");
-//    response.put("television 2", "LG");
-    return ResponseEntity.ok(televisionDataBase);
+  private final TelevisionService televisionService;
+
+  public TelevisionsController(TelevisionService televisionService) {
+    this.televisionService = televisionService;
   }
 
+  // GET
   @GetMapping("/televisions/{id}")
-  public ResponseEntity<Television> getTelevisionById(@PathVariable("id") int id) {
-    if (findTelevisionById(id) == null) {
-      throw new RecordNotFoundException();
+  public ResponseEntity<Television> getTelevisionById(@PathVariable int id) {
+    Optional<Television> television = televisionService.getTelevision(id);
+    if (television.isPresent()) {
+      return new ResponseEntity<>(television.get(), HttpStatus.OK);
     }
-    return ResponseEntity.ok(findTelevisionById(id));
+    return new ResponseEntity<>(HttpStatus.NOT_FOUND);
   }
 
+  @GetMapping("/televisions")
+  public ResponseEntity<List<Television>> getTelevisions() {
+    return ResponseEntity.ok().body(televisionService.getTelevisions());
+  }
+
+  // POST
   @PostMapping("/televisions")
-  public ResponseEntity<Integer> postTelevision(@RequestBody String television) {
-    if (television.length() > 20) {
-      throw new StringTooLongException("This string is way too long");
-    }
-    Television newTV = new Television(television);
-    televisionDataBase.add(newTV);
-    return ResponseEntity.status(HttpStatus.CREATED).body(newTV.getId());
+  public ResponseEntity<Object> addTelevision(@RequestBody Television television) {
+
+    isNameTooLong(television);
+    Television newTelevision = televisionService.addTelevision(television);
+    return ResponseEntity.status(HttpStatus.CREATED).body(newTelevision);

```

Comment:
Let er op dat je bij POST reqeusts ook de location header moet mee geven in de response. De URL die lijdt naar de "getById" van het zojuist gemaakt object. Als je de .created() methode gebruikt, dan kun je de URL meegeven in de parameter. Als je het op deze manier doet, dan kun je de .headers() methode gebruiken.